src/main/java/org/example/Operations/CollectionOperations.java

```diff
@@ -0,0 +1,254 @@
+     */
+     public Response create(String token, String lpath, CollectionsCreateParams prop) {
+         // contains parameters for the HTTP request
+         Map<Object, Object> formData = new HashMap<>();
+         formData.put("op", "create");
+         formData.put("lpath", lpath);
+         prop.getIntermediates().ifPresent(val -> formData.put("create-intermediates", String.valueOf(val)));
+
+         HttpResponse<String> response = HttpRequestUtil.sendAndParsePOST(formData, baseUrl, token, client.getClient());
+         return new Response(response.statusCode(), response.body());
+     }
+     public Response create(String token, String lpath) {
+         CollectionsCreateParams prop = new CollectionsCreateParams();
+         return this.create(token, lpath, prop);
+     }
+
+    /**
+     * Removes a collection
+     * Protected, so it can only be accessed from this package. Enforces use of builder
+     * @param lpath The logical path for the collection
+     * @param recurse If true, contents of the collection will be removed. Optional parameter
+     * @param noTrash If true, collection is permanently removed. Optional parameter
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public Response remove(String token, String lpath, CollectionsRemoveParams prop) {
+        // contains parameters for the HTTP request
+        Map<Object, Object> formData = new HashMap<>();
+        formData.put("op", "remove");
+        formData.put("lpath", lpath);
+        prop.getRecurse().ifPresent(val -> formData.put("recurse", String.valueOf(val)));
+        prop.getNoTrash().ifPresent(val -> formData.put("no-trash", String.valueOf(val)));
+
+        HttpResponse<String> response = HttpRequestUtil.sendAndParsePOST(formData, baseUrl, token, client.getClient());
+        return new Response(response.statusCode(), response.body());
+    }
+
+    public Response remove(String token, String lpath) {
+        CollectionsRemoveParams prop = new CollectionsRemoveParams();
+        return this.remove(token, lpath, prop);
+    }
+
+
+    /**
+     * Returns information about a collection
+     * @param lpath The logical path for the collection
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws IrodsException
+     */
+    public Response stat(String token, String lpath, CollectionsStatParams prop) {
+
+        // contains parameters for the HTTP request
+        Map<Object, Object> formData = new HashMap<>();
+        formData.put("op", "stat");
+        formData.put("lpath", lpath);
+        prop.getTicket().ifPresent(val -> formData.put("ticket", val));
+
+        HttpResponse<String> response = HttpRequestUtil.sendAndParseGET(formData, baseUrl, token, client.getClient());
+        return new Response(response.statusCode(), response.body());
+    }
+
+    public Response stat(String token, String lpath) {
+        CollectionsStatParams prop = new CollectionsStatParams();
```

Comment:
It's an overloaded method in case the user wants to use the `stat` method in its basic functionality, without any of the optional parameters.

Reply:
Hmm, couldn't the user of the library implement that themselves?

Reply:
I just figured it would be helpful to have the method with no parameters already implemented, but they user of the library definitely could implement it themselves.

Reply:
Let's stick with providing the base functionality for now. Convenience functions can always be added later. We'll learn more once people start using the library.

Reply:
Oh okay, it was only taking about 2 seconds to add but I can stop adding them moving forward.

Reply:
Should I remove the existing ones?

Reply:
I'm assuming you're referring to the convenience functions for some of the other operations? If that's true, then please remove those too.

Keep in mind you can use `Optional<T>` as parameters. This is good for functions which take one additional optional parameter.

Reply:
Okay, it's been removed.